multichannel_timeseries/1_large_multichan.ipynb

```diff
@@ -0,0 +1,954 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"admonition alert alert-warning\">\n",
+    "    <p class=\"admonition-title\" style=\"font-weight:bold\">Warning</p>\n",
+    "    Ensure you have sufficient disk space to handle the new data pyramid prior to running this section. If using the demo data, this will create a ~3 GB file on disk.\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "\n",
+    "# The FACTORS will depend on the size of your dataset, available disk size, and the resolution needed\n",
+    "# For this demo, we are arbitrarily choosing 8 levels and scaling by quadrupling each preceding factor.\n",
+    "FACTORS = [4**i for i in range(8)]\n",
+    "\n",
+    "def _help_downsample(data, time, n_out):\n",
+    "    \"\"\"\n",
+    "    Helper function for downsampling and returning as a specific format.\n",
+    "    \"\"\"\n",
+    "    indices = MinMaxLTTBDownsampler().downsample(time, data, n_out=n_out)\n",
+    "    return data[indices], indices\n",
+    "\n",
+    "\n",
+    "def apply_downsample(ts_ds, factor, dims):\n",
+    "    \"\"\"\n",
+    "    Apply downsampling to a time series dataset.\n",
+    "    \"\"\"\n",
+    "    dim = dims[0]\n",
+    "    n_out = len(ts_ds[\"data\"]) // factor\n",
+    "    print(f\"Downsampling by factor {factor} for a size of {n_out}.\")\n",
+    "    ts_ds_downsampled, indices = xr.apply_ufunc(\n",
+    "        _help_downsample,\n",
+    "        ts_ds[\"data\"],\n",
+    "        ts_ds[dim],\n",
+    "        kwargs=dict(n_out=n_out),\n",
+    "        input_core_dims=[[dim], [dim]],\n",
+    "        output_core_dims=[[dim], [\"indices\"]],\n",
+    "        exclude_dims=set((dim,)),\n",
+    "        vectorize=True,\n",
+    "        dask=\"parallelized\",\n",
+    "        dask_gufunc_kwargs=dict(output_sizes={dim: n_out, \"indices\": n_out}),\n",
+    "    )\n",
+    "    # Update the dimension coordinates with the downsampled indices\n",
+    "    ts_ds_downsampled[dim] = ts_ds[dim].isel(time=indices.values[0])\n",
+    "    return ts_ds_downsampled.rename(\"data\")\n",
+    "\n",
+    "\n",
+    "if not PYRAMID_PATH.exists():\n",
+    "    print(f'Creating Pyramid: {PYRAMID_PATH}')\n",
+    "    ts_dt = pyramid_create(\n",
+    "        ts_ds,\n",
+    "        factors=FACTORS,\n",
+    "        dims=[\"time\"],\n",
+    "        func=apply_downsample,\n",
+    "        type_label=\"pick\",\n",
+    "        method_label=\"pyramid_downsample\",\n",
+    "    )\n",
+    "    display(ts_dt)\n",
+    "else:\n",
+    "    print(f'Pyramid Already Exists: {PYRAMID_PATH}')"
+   ]
```

Comment:
I think the bottom of this cell is a good place to add `f.close()`.

Reply:
f is only used in conversion to xarray. I added a context manager instead of f.close()

Reply:
Arf now I get this error when the pyramid gets created, while I don't when `close` is called afterwards. I'll make the change.

```
OSError: Can't synchronously read data (dset_id is not a dataset ID)
```